src/polyfem/mesh/SlimSmooth.cpp

```diff
@@ -0,0 +1,134 @@
+#include "SlimSmooth.hpp"
+
+#include <igl/boundary_facets.h>
+#include <polyfem/utils/Logger.hpp>
+#include <igl/slim.h>
+
+namespace polyfem::mesh
+{
+	namespace
+	{
+		double triangle_jacobian(const Eigen::VectorXd &v1, const Eigen::VectorXd &v2, const Eigen::VectorXd &v3)
```

Comment:
These functions also appear in AdjointNLProblem, can be moved to a header

Reply:
done